sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -76,7 +76,15 @@ case class CurrentTimestamp() extends LeafExpression with CodegenFallback {
  */
 @ExpressionDescription(
   usage = "_FUNC_(start_date, num_days) - Returns the date that is num_days after start_date.",
-  extended = "> SELECT _FUNC_('2016-07-30', 1);\n '2016-07-31'")
+  extended = """
+    Arguments:
+      start_date - a date expression.
+      num_days - a numeric expression that represents the number of days to add.
```

Comment:
This one allows implicit cast. So, actually `date_add('2016-07-30', 1.0)` works fine. Also, it takes expressions.

```scala
scala> spark.range(10).selectExpr("date_add('2016-07-30', id)").show()
+---------------------------------------------------+
|date_add(CAST(2016-07-30 AS DATE), CAST(id AS INT))|
+---------------------------------------------------+
|                                         2016-07-30|
|                                         2016-07-31|
|                                         2016-08-01|
|                                         2016-08-02|
|                                         2016-08-03|
|                                         2016-08-04|
|                                         2016-08-05|
|                                         2016-08-06|
|                                         2016-08-07|
|                                         2016-08-08|
+---------------------------------------------------+
```